public/pages/Detectors/containers/Detectors/Detectors.tsx

```diff
@@ -122,8 +122,16 @@ export default class Detectors extends Component<DetectorsProps, DetectorsState>
     } catch (e: any) {
       errorNotificationToast(notifications, 'update', 'detector', e);
     }
-    this.getDetectors();
-    this.setState({ loadingDetectors: false });
+    await this.getDetectors();
+    const selectedItemIds = new Set(this.state.selectedItems.map(({ _id }) => _id));
+    const updatedSelectedItems: DetectorHit[] = [];
+    this.state.detectorHits.forEach((hit) => {
```

Comment:
Minor: can use filter

`const updatedSelectedItems = this.state.detectorHits.filter(hit => selectedItemIds.has(hit._id));`

to include only those items whose _id is in selectedItemIds

Reply:
Thanks for catching, let me change this